lib/jarvis/thread_logger.rb

```diff
@@ -0,0 +1,42 @@
+require "cabin"
+
+module Jarvis module ThreadLogger
+  class State
+    def initialize
+      @state = {}
+      @lock = Mutex.new
+    end
+
+    def <<(hash)
+      @lock.synchronize do
+        @state[Thread.current] = hash
```

Comment:
Yeah sorry; I wasn't clear ;P

This is for being able to ask Jarvis what each pool worker is currently doing.